usefulUtilities/statusIndicator/webApp/app/statusIndicator_webApp.js

```diff
@@ -0,0 +1,325 @@
+//
+// statusIndicator_webApp.js
+//
+// NodeJS Web App for Status Indicator App
+// Created by Robin Wilson on 2019-04-02
+// Copyright High Fidelity 2019
+//
+// Licensed under the Apache 2.0 License
+// See accompanying license file or http://apache.org/
+
+var http = require('http');
+var url = require('url');
+var dbInfo = require('./dbInfo.json');
+
+
+// Heartbeat updates setTimeout and updates employee data in database
+var users = {}; // { username: { timer: <timer object>} }
+function heartbeat(queryParamObject, response) {
+    var username = queryParamObject.username;
+    // handle heartbeat
+    if (!users[username]) {
+        // new heartbeat for username
+        users[username] = {};
+    } else {
+        // stop the old heartbeat timer
+        clearTimeout(users[username].timer);
+    }
+    users[username].timer = startHeartbeatTimer(username, response);
+
+    if (username) {
+        updateEmployee(queryParamObject, response);
+    } else {
+        console.error("There was no username or primary key! Can't update or create employee!");
+        return;
+    }
+}
+
+
+// If timer runs out, will set user to offline
+var HEARTBEAT_INTERVAL_MS = 10000;
+function startHeartbeatTimer(username, response) {
+    var heartbeatTimer = setTimeout(function() {
+        // user has stopped sending heartbeat()
+        // update user to offline
+        users[username].timer = null;
+        updateEmployee({
+            username: username,
+            status: "offline"
+        }, response);
+    }, HEARTBEAT_INTERVAL_MS);
+    return heartbeatTimer;
+}
+
+
+// Update employee status in database
+function updateEmployee(updates, response) {
+    // Create strings for query from the updates object
+    var columnString = ""; // (username, displayName, status)
+    var valueString = ""; // (username1, Display Name, busy)
```

Comment:
These have to be in quotes, right?
```suggestion
    var valueString = ""; // ("my_username", "My Display Name", "busy")
```

Reply:
yep!